### ISSUE TITLE
twisted.conch.telnet should be ported to Python 3

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8228]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)| @hawkowl reported|
|-|-|
|Trac ID|trac#8228|
|Type|enhancement|
|Created|2016-03-08 05:24:04Z|
|Branch|https://github.com/twisted/twisted/tree/telnet-py3-8228|

Required modules:

- twisted.conch.telnet
- twisted.conch.test.test_telnet
- twisted.protocols.telnet

<details><summary>Searchable metadata</summary>

```
trac-id__8228 8228
type__enhancement enhancement
reporter__hawkowl hawkowl
priority__normal normal
milestone__Python_3_x Python-3.x
branch__branches_telnet_py3_8228 branches/telnet-py3-8228
branch_author__hawkowl hawkowl
status__closed closed
resolution__fixed fixed
component__conch conch
keywords__None None
time__1457414644416527 1457414644416527
changetime__1457602348631921 1457602348631921
version__None None
owner__hawkowl hawkowl
cc__z3p
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_2"></a>|@hawkowl commented|
|-|-|

(In [46943]) Branching to telnet-py3-8228.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_4"></a>|@hawkowl commented|
|-|-|

I'm happy with this. Builds are green enough. Please review.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_5"></a>|@adiroiban set owner to @hawkowl|
|-|-|


Minor comments:

1 Please see

```
************* Module twisted.conch.test.test_telnet
W9202:670,4:TelnetTests._deliver: Missing epytext markup @param for argument "b"
************* Module twisted.python.compat
W9202:690,0:_bytesChr: Missing epytext markup @param for argument "i"
```

2 Why the private _bytesChr in compat ?

Looks good for the initial port.

I am not sure how popular is conch.telnet. I have never used it and I am not sure how this will interact with Unicode support in telnet and if conch.telnet support that.

Please check my comments and merge.

If you know a conch.telnet user which can review this, please put it back for review.

Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_6"></a>|@hawkowl commented|
|-|-|

Hi Adi, thanks for the review.

1. Fixed them up a bit.

2. chr() on Py3 returns Unicode strings, and it was used on Python 2 to get a byte with the value it was given. I've improved the docstring.

As far as popularity, I'm not porting this because it by itself is useful, but because twistd manhole + a few other things use telnet, which makes it a simple porting prerequisite.

Will merge.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_7"></a>|@hawkowl set status to `closed`|
|-|-|

(In [46968]) Merge telnet-py3-8228: Port twisted.conch.telnet to Python 3

Author: hawkowl
Reviewer: adiroiban
Fixes: [#8228](https://github.com/twisted/twisted/issues/8228)
